falcon/api.py

```diff
@@ -786,22 +787,21 @@ def _handle_exception(self, req, resp, ex, params):
             bool: ``True`` if a handler was found and called for the
             exception, ``False`` otherwise.
         """
+        err_handler = self._find_error_handler(ex)
 
-        for err_type, err_handler in self._error_handlers:
-            if isinstance(ex, err_type):
-                try:
-                    err_handler(req, resp, ex, params)
-                except HTTPStatus as status:
-                    self._compose_status_response(req, resp, status)
-                except HTTPError as error:
-                    self._compose_error_response(req, resp, error)
-
-                return True
-
-        # NOTE(kgriffs): No error handlers are defined for ex
-        # and it is not one of (HTTPStatus, HTTPError), since it
-        # would have matched one of the corresponding default
-        # handlers.
+        if err_handler is not None:
+            try:
+                err_handler(req, resp, ex, params)
+            except HTTPStatus as status:
+                self._compose_status_response(req, resp, status)
+            except HTTPError as error:
+                self._compose_error_response(req, resp, error)
+
+            return True
+
+        # NOTE(kgriffs): No error handlers are defined for ex and it
+        # is not one of (HTTPStatus, HTTPError, Exception), since it
```

Comment:
I realized that this note didn't get updated in #1527. I guess it is not too late to add that change there, but I think doing so will override the approval?

Reply:
I was the only who approved, I can do it again :slightly_smiling_face: 
Not too late IMHO.

Reply:
Heh. Funny how after months and months of that PR sitting there unchanged, I was too late to squeeze in one more change because I left my laptop alone for 4 hours :joy:

Reply:
Sorry, I guess I jumped the gun. 😄

Reply:
No worries! I did say that it was ready to go.